specs/TransciptTestRunner.md

```diff
@@ -37,6 +40,59 @@ Here is a high level class diagram to get started
 - `_XYZ_Client`: specialized client that knows how to interact with a specific channel (it should be capable of generating channel OAuthTokens, handle responses coming from that specific channel, etc.)
 - `ClientType` (enum): types are: DirectLine, Emulator, Teams
 
+## TestRunner Class
+### Methods
+- `RunTestAsync(string)`  Executes a single scenario based on the transcript passed to the method.   
+    * _Parameters_   
+        * transcript `string`    The transcript that contains the scenario to execute.  
+    * _Returns_  
+A `Task`.    
+
+    * Sample Usage   
+<img src="media/runtestasync1.png" width="700" border=1px>
+- `RunTestAsync(IEnumerable<string>) ` 
+Executes a test based on a collection of transcripts, in sequence, in the order in which they exist in the collection. This method can be used to compose a complex scenario formed by a set of smaller scenarios defined in each transcript.  
+    * _Parameters_   
+        * transcripts `IEnumerable<string>` The collection of transcripts to execute.  
+    * _Returns_  
+A `Task`.   
+
+    * Sample Usage    
+<img src="media/runtestasync2.png" width="700" border=1px>
+- `RunTestAsync(string, params string[])`  Executes a scenario utilizing a parameterized transcript.    
+    * _Parameters_      
+        * transcript `string`    The parameterized transcript that contains the scenario to execute.  
+        * values `params string[]`    The parameters to be passed to the transcript.    
+    * _Returns_  
+A `Task`.  
+    * Sample Usage  
+<img src="media/runtestasync3.png" width="700" border=1px>
+- `SendActivityAsync(Activity)`  Sends a single Activity to the test bot.    
+    * _Parameters_    
+        * activity `Microsoft.Bot.Schema.Activity`  The Activity that is sent to the bot.    
+    * _Returns_
+A `Task<bool>`
+- `ContinueWithAsync(string)` Executes a transcript as a continuation to the last scenario.
```

Comment:
I don't think we need this, it may be easier to call RunTestAsync multiple times, based on what I see in the code right now.